src/main/java/org/ethereum/beacon/discovery/DiscoverySystemBuilder.java

```diff
@@ -90,14 +93,19 @@ public DiscoverySystem build() {
     final int clientNumber = COUNTER.incrementAndGet();
     final LocalNodeRecordStore localNodeRecordStore =
         new LocalNodeRecordStore(localNodeRecord, privateKey);
+    final ExpirationSchedulerFactory expirationSchedulerFactory =
+        new ExpirationSchedulerFactory(
+            Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder().setNameFormat("discovery-expiration-%d").build()));
     final DiscoveryManager discoveryManager =
         new DiscoveryManagerImpl(
             nodeTable,
             nodeBucketStorage,
             localNodeRecordStore,
             privateKey,
             nodeRecordFactory,
-            schedulers.newSingleThreadDaemon("client-" + clientNumber));
+            schedulers.newSingleThreadDaemon("client-" + clientNumber),
```

Comment:
nit: Would it be better to have a Thread name that gives more context into what the thread is doing? 
```suggestion
            schedulers.newSingleThreadDaemon("client-expiration-scheduler-" + clientNumber),
```

Reply:
Good idea, but that's not the expiration schedule thread, that line is unchanged except for the `,` at the end so we could pass the `expirationSchedulerFactory` through.  It's thread is `discovery-expiration-%d`.  I've prepended them with `discovery-` to make it clearer.